src/Vector.cpp

```diff
@@ -0,0 +1,74 @@
+#include "Vector.h"
+#include <algorithm>
+
+Vector::Vector(size_t initial_size, int initial_value) : m_size(initial_size), m_capacity(initial_size) {
+    m_data = new int[m_capacity];
```

Comment:
Also, what if `initial_size==0`? This will allocate a 0 size array which is undefined